DeviceAdministration/Infrastructure.UnitTests/Infrastructure/IoTHubRepositoryTests.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.Devices.Applications.RemoteMonitoring.Common.Models;
+using Microsoft.Azure.Devices.Applications.RemoteMonitoring.DeviceAdmin.Infrastructure.Repository;
+using Moq;
+using Ploeh.AutoFixture;
+using Xunit;
+
+namespace Microsoft.Azure.Devices.Applications.RemoteMonitoring.DeviceAdmin.Infrastructure.UnitTests.Infrastructure
+{
+    public class IoTHubRepositoryTests
+    {
+        private readonly IIotHubRepository iotHubRepository;
+        private readonly Mock<IDeviceManager> deviceManagerMock;
+        private readonly Fixture fixture;
+
+        public IoTHubRepositoryTests()
+        {
+            this.deviceManagerMock = new Mock<IDeviceManager>();
+            this.deviceManagerMock.Setup(dm => dm.AddDeviceAsync(It.IsAny<Device>())).Returns(Task.FromResult(new Device()));
+            this.deviceManagerMock.Setup(dm => dm.RemoveDeviceAsync(It.IsAny<string>())).Returns(Task.FromResult(true));
+            this.iotHubRepository = new IotHubRepository(this.deviceManagerMock.Object);
+            this.fixture = new Fixture();
+        }
+
+        [Fact]
+        public async void AddDeviceAsync()
+        {
+            var device = this.fixture.Create<DeviceModel>();
+            var keys = new SecurityKeys("fbsIV6w7gfVUyoRIQFSVgw ==", "1fLjiNCMZF37LmHnjZDyVQ ==");
+            var sameDevice = await this.iotHubRepository.AddDeviceAsync(device, keys);
+            Assert.Equal(sameDevice, device);
+        }
+
+        [Fact]
+        public async void TryAddDeviceAsync()
+        {
+            var device = new Device("deviceId")
+                         {
+                             Authentication = null,
+                             Status = new DeviceStatus()
+                         };
+
+            var result = await this.iotHubRepository.TryAddDeviceAsync(device);
+            Assert.True(result);
+
```

Comment:
IoTHubRepository doesn't do any validation for device objects. We can test that scenario for a class that uses DeviceSchemaHelper and send invalid device objects.